Sources/MongoMobile/MongoMobile.swift

```diff
@@ -81,11 +68,22 @@ public class MongoMobile {
     private static var embeddedInstances = [String: OpaquePointer]()
     /// Cache embedded clients for cleanup
     private static var embeddedClients = [WeakRef<MongoClient>]()
+    /// Store user-provided logger.
+    internal static var logger: MongoMobileLogger?
 
     /**
      * Perform required operations to initialize the embedded server.
+     * 
+     * Parameters:
+     *  - logger: An optional `MongoMobileLogger` to use with the embedded server.
+     *            This logger will be used across all DBs and clients until the 
+     *            embedded server is shut down via `close()`.
+     *
+     * Throws:
+     *  - `MongoMobileError.invalidInstance` if there is any error initializing 
+     *    the embedded server.
      */
-    public static func initialize() throws {
+    public static func initialize(withLogger logger: MongoMobileLogger? = nil) throws {
```

Comment:
I think we should create an options struct, even if just for a single field for now. Something like:
```
public static func initialize(_ options: MongoMobileOptions? = nil) throws {
```

Was there a reason you went with this approach that I'm not considering? Seems like options aligns better with the embedded API, and allows us more flexibility for future extension.